turbo/engineapi/engine_server.go

```diff
@@ -442,7 +445,12 @@ func (s *EngineServer) forkchoiceUpdated(ctx context.Context, forkchoiceState *e
 
 		status, err = s.HandlesForkChoice("ForkChoiceUpdated", forkchoiceState, 0)
```

Comment:
I suggest to change `HandlesForkChoice` to return `(ForkChoiceReceipt, error)`. Thus we can use `ForkChoiceReceipt.ValidationError` for validation errors and keep the old logic where `err` was meant for internal errors like MDBX ones rather than validation errors.